pythonforandroid/bootstraps/common/build/build.py

```diff
@@ -336,9 +336,11 @@ def make_package(args):
         with open(args.intent_filters) as fd:
             args.intent_filters = fd.read()
 
-    # if get_bootstrap_name() == "sdl2":
-    args.add_activity = args.add_activity or []
-    args.activity_launch_mode = args.activity_launch_mode or ''
+    if not args.add_activity:
```

Comment:
Ternary is still possible if you like:
```
args.add_activity = args.add_activity if args.add_activity else []
```

Reply:
@AndreMiras doesn't that work in the same way though? won't `or` return the value to the right if the left-hand side would also evaluate in a conditional to false? (which would then give the same result as your ternary operator) or is there an actual semantic difference in what it produces?

While I agree it's not a ternary operator, I don't see why that is a problem. Is it specifically discouraged or something? I've seen `X or Y` quite often so it does seem to be a common pattern

Reply:
Yes there is totally a semantic difference. I also see it very often, but most of the time people use it as they think it's ternary (because other languages follow this pattern).
But yes it's semantically different, but I suspect we wanted a ternary and not this

Reply:
I know, I prefer shorten lines, also I thing that is more consistent with the module code like I did, but if you want I will change it, do you want that? 
;)

Reply:
I don't really mind @opacam :+1:

Reply:
My vote is to keep it as `args.add_activity = args.add_activity or []`. I don't see any benefit in turning this into a ternary operator

Reply:
Aaa..we can do the same in here:

`args.add_activity = getattr(args, 'add_activity', [])`

and

`args.activity_launch_mode = getattr(args, 'activity_launch_mode', '')`

Reply:
That doesn't do the same thing. `getattr` returns the value if the attribute *isn't present*, while `X or Y` will return `Y` if `X` is None and error if the attribute isn't present.

Reply:
Yes my bad, I got mislead by other things I've seen too often, but it involves the `and` keyword like here https://github.com/kivy/python-for-android/blob/dab2a8a/pythonforandroid/bootstraps/pygame/build/buildlib/jinja2.egg/jinja2/filters.py#L300
Thanks for the head up!